### PR TITLE
Release 6.0.0

### PR DESCRIPTION
**Pull request by courtneycb**
**Changelog:**

- Add 'Plugging it in' area of the website:
    - Includes 21 programming challenges in Python for Binary Numbers.
    - Includes 9 programming challenges in Python for Kidbots.
    - Saves a users code attempt and their status on the question.
    - User code tested on the JOBE server.
    - Scratch questions are not supported.
- Adds learning outcome and solution content to programming challenges table in the educators area.
- Replace content under the Python dropdown on programming challenge pages in CSU with a link to the same challenge in Plugging it in.
- Order glossary terms alphabetically for all languages.
- Solutions provided on the standard CSU site now pass the tests for the respective programming challenge on the CSU Plugging it in site.
- Host videos on Vimeo instead of YouTube.
- Edit the formatting of subtitle files for Vimeo.
- Put testing examples for programming challenges into a separate markdown file.
- Add Google Tag Manager.
- Minor content fixes.

### CODE REVIEW

**Review by eAlasdair — COMMENTED** [review]
As of 282/512 files viewed

File: csunplugged/static/js/test-code.js [hunk 1]
```diff
@@ -78,15 +79,23 @@ async function run_testcase(
     } else {
       testcaseResult.status = "Failed";
       testcaseResult.userOutput = userResult.stdout;
+      testcaseResult.helpInfo = "Looks like your code was run successfully but didn’t match the expected output. Have a look at the ‘Expected Output’ and the 'Received Output’. See if you can spot the differences!"
```
[the remark this answers is not in the document]
Comment (eAlasdair) [reply csunplugged/static/js/test-code.js:82]: Oh that needs #1434 first

File: csunplugged/topics/content/en/binary-numbers/programming-challenges/count-5-bw-using-loop/python-hints.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 -   Make another variable called `number_of_dots` and set its value to 16.
 -   Add the value of `number_of_dots` to the value of `total_number_of_dots`
     every time the end user enters ‘W’ as the input.
-    Repeat this 5 times dividing the value of `number_of_dots` by 2 each time.
+    Repeat this 5 times dividing the value of `number_of_dots` by 2 each time and
+    round the result to the nearest integer using `round()`.
```
[the remark this answers is not in the document]
Comment (eAlasdair) [reply csunplugged/topics/content/en/binary-numbers/programming-challenges/count-5-bw-using-loop/python-hints.md:6]: I now think 'convert it to an integer using int()' is best. round implies it's not a whole number (everything _except_ the result of n.0), but int just implies its not an integer (everything including the result)

**Comment by eAlasdair** [comment]
https://www.vimeo.com/uccser is permanently redirected to https://vimeo.com/uccser so that should be changed (not sure where exactly that is)

File: docs/source/changelog.rst [hunk 3]
```diff
@@ -23,6 +23,49 @@ All notable changes to this project will be documented in this file.
...
+**Changelog:**
+
+- Add 'Plugging it in' area of the website:
+    - Includes 21 programming challenges in Python for Binary Numbers.
+    - Includes 9 programming challenges in Python for Kidbots.
+    - Saves a users code attempt and their status on the question.
+    - User code tested on the JOBE server.
+    - Scratch questions are not supported.
+    - User triggered walk-through on programming challenge page. 
+- Adds learning outcome and solution content to programming challenges table in the educators area.
```
Comment (eAlasdair) [inline docs/source/changelog.rst:45]:
```suggestion
- Add learning outcome and solution content to programming challenges table in the educators area.
```

File: csunplugged/templates/topics/programming-challenges-table.html [hunk 4]
```diff
@@ -2,41 +2,100 @@
...
+</div>
+
+{% block scripts %}
+  <script type="text/javascript">
+    function updateToggleButtonText(challenge_num) {
+      challengeContentToggleId = `#toggle-${challenge_num}`;
+      if ($(challengeContentToggleId).hasClass('collapsed')) {
+        $(challengeContentToggleId).text('Show less details');
+      } else {
+        $(challengeContentToggleId).text('Show more details');
```
Comment (eAlasdair) [inline csunplugged/templates/topics/programming-challenges-table.html:96]: If two challenges have the same ID number (see [How Binary Digits Work](https://cs-unplugged-dev.appspot.com/en/topics/binary-numbers/unit-plan/how-binary-digits-work/programming/)) Both sets of outcomes are expanded on one button push

**Review by eAlasdair — CHANGES_REQUESTED** [review]
Just 1 significant problem I see

File: csunplugged/templates/topics/programming-challenges-table.html (hunk 4 above)
Comment (courtneycb) [reply csunplugged/templates/topics/programming-challenges-table.html:96]: Awesome spot

File: csunplugged/static/js/plugging-it-in-walkthrough.js [hunk 5]
```diff
@@ -29,12 +29,12 @@ $(document).ready(function () {
...
             {
                 element: "#test-case-table",
-                intro: "These are the test cases that will be run against your program.",
+                intro: "These are the test cases that will be used to see if your program is working correctly.",
                 step: 5
             },
             {
                 element: "#test-case-table tbody tr:nth-child(1)",
-                intro: "Here is the first test case.",
+                intro: "This box shows what happened for the first test case.",
```
Comment (eAlasdair) [inline csunplugged/static/js/plugging-it-in-walkthrough.js:37]: This is in past tense, unlike every other box. But with a change to 'happens' it doesn't quite make sense. I'm a fan of leaving it to what it was

**Review by eAlasdair — APPROVED** [review]
Hooray! :tada: :party_parrot:

**Pull request merged**